sdk/python/approzium/_authenticator.py

```diff
@@ -28,16 +28,55 @@ class AuthClient(object):
 
     :param server_address: address (host:port) at which an authenticator
         service is listening.
-
     :type server_address: str
+
+    :param disable_tls: defaults to False. When False, https is used and a
+        client_cert and client_key proving the client's identity must be
+        provided. When True, http is used and no other TLS options must be
+        set.
+    :type disable_tls: bool, optional
+
+    :param trusted_certs: the path to one or many identity certificates that
+        the Approzium authenticator may present.
+    :type trusted_certs: str, optional
+
+    :param client_cert: the path to this client's identity certificate.
+        Required when disable_tls is False.
+    :type client_cert: str, optional
+
+    :param client_key: the path to the private key this client used to obtain
+        its identity certificate. Required when disable_tls is False.
+    :type client_key: str, optional
+
     :param iam_role: if an IAM role Amazon resource number (ARN) is provided,
         it will be assumed and its identity will be used for authentication.
         Otherwise, the default ``boto3`` session will be used as the identity.
     :type iam_role: str, optional
     """
 
-    def __init__(self, server_address, iam_role=None):
+    def __init__(
+        self,
+        server_address,
+        disable_tls=False,
+        trusted_certs=None,
+        client_cert=None,
+        client_key=None,
+        iam_role=None,
+    ):
         self.server_address = server_address
+
+        if not disable_tls:
+            if client_cert is None or client_key is None:
+                raise ValueError(
+                    "if tls is not disabled, client_cert and client_key are required"
+                )
+
+        self.tls_config = TLSConfig(
+            disable_tls=disable_tls,
+            trusted_certs=trusted_certs,
```

Comment:
Amazing work Becca 😄 . It looks like the TLSConfig class is just for storing TLS-related parameters. If the user is going to pass the TLS-related parameters *separately* anywyay, then I don't think we need that class, or at least we can make it internal, since we don't expect the user to instantiate it.

Instead, I actually suggest that we change the interface such that if the user wants to enable TLS, they can pass in a TLSConfig instance when they instantiate the AuthClient. If they want to disable TLS, they can use disable_tls=True and not pass in a TLSConfig.

Reply:
I added this in https://github.com/cyralinc/approzium/pull/111/commits/5a949ac2f0edb8885e7484703265634e369ba002! Let me know if it doesn't look right. :-)